File: internal/screens/results.go
package screens

import (
	"fmt"
	"sort"

	tea "charm.land/bubbletea/v2"

	"github.com/jashkarangiya/valo-tui/internal/data"
	"github.com/jashkarangiya/valo-tui/internal/widgets"
)

var resultOrder = map[string]int{"live": 0, "completed": 1}

// Results is the [r] event sub-page: completed & live series. Enter drills into
// the per-map scoreboards.
type Results struct {
	w, h   int
	table  widgets.Table
	hasEvt bool
}

func NewResults(w, h int) Results {
	return Results{w: w, h: h, table: widgets.NewTable(
		widgets.Column{Title: "status", Width: 6},
		widgets.Column{Title: "team 1", Width: 18},
		widgets.Column{Title: "score", Width: 5},
		widgets.Column{Title: "team 2", Width: 18},
		widgets.Column{Title: "phase", Width: 15},
		widgets.Column{Title: "when", Width: 12},
	)}
}

func (s *Results) SetSize(w, h int) {
	s.w, s.h = w, h
	s.table.SetHeight(h - 4)
}

func (s *Results) Load(eventID int, eventName string) {
	s.hasEvt = eventID != 0
	if !s.hasEvt {
		s.table.SetRows(nil)
		return
	}
	matches := []data.MatchCard{}
	for _, m := range data.EventMatchCards(eventID, eventName) {
		if _, ok := resultOrder[m.Status]; ok {
			matches = append(matches, m)
		}
	}
	sort.SliceStable(matches, func(a, b int) bool {
		return resultOrder[matches[a].Status] < resultOrder[matches[b].Status]
	})
	rows := make([]widgets.Row, 0, len(matches))
	for _, m := range matches {
		status := widgets.Cell{Text: "✓ done", Style: mutedSt}
		scoreStyle := mutedSt
		if m.IsLive() {
			status = widgets.Cell{Text: "● live", Style: liveSt}
			scoreStyle = liveSt
		}
		s1, s2 := "–", "–"
		if m.Team1.Score != nil {
			s1 = fmt.Sprint(*m.Team1.Score)
		}
		if m.Team2.Score != nil {
			s2 = fmt.Sprint(*m.Team2.Score)
		}
		when := m.Time
		if when == "" {
			when = m.Date
		}
		// Dim the losing side of a completed match so the winner reads at a glance.
		t1Style, t2Style := textSt, textSt
		if m.Status == "completed" {
			if a, b := derefOr0(m.Team1.Score), derefOr0(m.Team2.Score); a > b {
				t2Style = mutedSt
			} else if b > a {
				t1Style = mutedSt
			}
		}
		rows = append(rows, widgets.Row{
			Key: fmt.Sprint(m.MatchID),
			Cells: []widgets.Cell{
				status,
				{Text: m.Team1.Name, Style: t1Style},
				{Text: s1 + "–" + s2, Style: scoreStyle},
				{Text: m.Team2.Name, Style: t2Style},
				{Text: m.Phase, Style: mutedSt},
				{Text: when, Style: mutedSt},
			},
		})
	}
	s.table.SetRows(rows)
}

func (s *Results) Focus()                           { s.table.Focus() }
func (s *Results) Blur()                            { s.table.Blur() }
func (s Results) Selected() string                  { return s.table.SelectedKey() }
func (s *Results) ClickVisual(i int) (string, bool) { return s.table.ClickVisual(i) }
func (s *Results) MoveCursor(d int)                 { s.table.MoveCursor(d) }

func (s Results) Update(msg tea.Msg) (Results, tea.Cmd) {
	if k, ok := msg.(tea.KeyPressMsg); ok {
		tableMove(&s.table, k.String())
	}
	return s, nil
}

func (s Results) View() string {
	header := title("results") + "\n" + hint("j/k move · enter → scoreboards") + "\n\n"
	if !s.hasEvt {
		return header + muted("select an event first")
	}
	if s.table.Len() == 0 {
		return header + muted("no completed matches yet")
	}
	return header + s.table.View()
}

File: valo_tui/screens/results.py
"""[r] results — completed and live series for the focused event.

Event-scoped (never global): the rows come from the event's own match list.
Pressing Enter on a row drills into the per-map scoreboards."""

from __future__ import annotations

from rich.text import Text

from textual.app import ComposeResult
from textual.containers import Vertical
from textual.widgets import Label

from ..data import cache
from ..data.models import MatchCard
from .widgets import LIVE, MUTED, TEXT, VimDataTable

# Live first, then most-recently completed.
_ORDER = {"live": 0, "completed": 1}


class ResultsView(Vertical):
    def compose(self) -> ComposeResult:
        yield Label("results", classes="page-title")
        yield Label("j/k move · enter → scoreboards", classes="hint")
        yield VimDataTable(id="results-table", cursor_type="row", zebra_stripes=False)

    def on_mount(self) -> None:
        t = self.query_one(VimDataTable)
        t.add_column("status", width=7, key="status")
        t.add_column("match", width=34, key="match")
        t.add_column("score", width=7, key="score")
        t.add_column("phase", width=20, key="phase")
        t.add_column("when", key="when")
        self.load_data()

    def load_data(self) -> None:
        t = self.query_one(VimDataTable)
        prev = t.cursor_row
        t.clear()
        eid = getattr(self.app, "current_event_id", None)
        if eid is None:
            t.add_row(Text("select an event first", style=MUTED),
                      Text(""), Text(""), Text(""), Text(""))
            return

        matches = [
            m for m in cache.event_match_cards(eid, self.app.event_name)
            if m.status in _ORDER
        ]
        matches.sort(key=lambda m: _ORDER.get(m.status, 9))
        for m in matches:
            t.add_row(*self._row(m), key=str(m.match_id))
        if matches:
            t.move_cursor(row=min(prev, len(matches) - 1))
        else:
            t.add_row(Text("no completed matches yet", style=MUTED),
                      Text(""), Text(""), Text(""), Text(""))

    def _row(self, m: MatchCard) -> tuple[Text, ...]:
        if m.is_live:
            status = Text("● live", style=LIVE)
        else:
            status = Text("✓ done", style=MUTED)
        match = Text(f"{m.team1.name}  vs  {m.team2.name}", style=TEXT)
        s1 = m.team1.score if m.team1.score is not None else "–"
        s2 = m.team2.score if m.team2.score is not None else "–"
        score = Text(f"{s1}–{s2}", style=LIVE if m.is_live else MUTED)
        phase = Text(m.phase, style=MUTED)
        when = Text(m.time or m.date or "", style=MUTED)
        return status, match, score, phase, when
